server/sonar-process/src/main/java/org/sonar/process/NetworkUtils.java

```diff
@@ -22,29 +22,29 @@
 import java.io.IOException;
 import java.net.Inet4Address;
 import java.net.InetAddress;
-import java.net.InetSocketAddress;
 import java.net.NetworkInterface;
 import java.net.ServerSocket;
 import java.net.SocketException;
 import java.net.UnknownHostException;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.stream.Collectors;
-import org.apache.commons.io.IOUtils;
-import org.apache.commons.lang.ArrayUtils;
 
 import static java.lang.String.format;
 import static java.util.Collections.list;
 import static org.apache.commons.lang.StringUtils.isBlank;
 
 public final class NetworkUtils {
 
-  private static final RandomPortFinder RANDOM_PORT_FINDER = new RandomPortFinder();
+  private static final Set<Integer> ALREADY_ALLOCATED = new HashSet<>();
```

Comment:
This set will fill during all the tests in the same JVM so even if a port is available we won't be able to find an available port if the method is called more than ~64512 not a big deal for the time being.